repo/content/content_upgrade_lock.go

```diff
@@ -0,0 +1,160 @@
+package content
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// UpgradeLock represents the intent to lock a kopia repository for upgrade
+// related maintenance activity. This signals a request for exclusive access to
+// the repository. The lock object is set on the Kopia repository format blob
+// 'kopia.repository' and must be respected by all clients accessing the
+// repository.
+type UpgradeLock struct {
+	OwnerID                string        `json:"ownerID,omitempty"`
+	CreationTime           time.Time     `json:"creationTime,omitempty"`
+	AdvanceNoticeDuration  time.Duration `json:"advanceNoticeDuration,omitempty"`
+	IODrainTimeout         time.Duration `json:"ioDrainTimeout,omitempty"`
+	StatusPollInterval     time.Duration `json:"statusPollInterval,omitempty"`
+	OldFormatVersion       FormatVersion `json:"oldFormatVersion,omitempty"`
+	MaxPermittedClockDrift time.Duration `json:"maxPermittedClockDrift,omitempty"`
+}
+
+// NewUpgradeLock constructs a new lock object for repository format upgrades.
+func NewUpgradeLock(
```

Comment:
does this function add much value? by calling it we're effectivelly hiding the meaning of various parameters. I would prefer explicit initialization + validate

Reply:
Constructor style pattern is what I prefer, but I'd avoid having a mix of coding patterns in a single project. Consistency is more important, so would switch to init + validate pattern if that is what we follow all across Kopia.